src/ValueParsers/MatrixParser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace ValueParsers;
+
+use DataValues\DecimalValue;
+use DataValues\IllegalValueException;
+use DataValues\MatrixValue;
+use InvalidArgumentException;
+
+/**
+ * ValueParser that parses the string representation of a matrix.
+ *
+ * @licence GNU GPL v2+
+ * @author Andrius Merkys < andrius.merkys@gmail.com >
+ */
+class MatrixParser extends StringValueParser {
+
+	const FORMAT_NAME = 'matrix';
+
+	/**
+	 * @see StringValueParser::stringParse
+	 *
+	 * @param string $value
+	 *
+	 * @return MatrixValue
+	 * @throws ParseException
+	 */
+	protected function stringParse( $value ) {
+		if ( !is_string( $value ) ) {
+			throw new InvalidArgumentException( '$value must be a string' );
+		}
+
+		$decoded = json_decode( $value );
```

Comment:
JSON is not exactly a human friendly format... ValueParsers are for human input. It seems kind of pointless to even have one for DataValues that cannot conveniently be edited as a single line string.

Hm, actually, there is nothing that forces this to be single line. It could take a multi-line representation of the matrix.